rules/TypeDeclaration/NodeAnalyzer/CallTypesResolver.php

```diff
@@ -67,7 +67,7 @@ public function resolveTypesFromCalls(array $calls): array
         foreach ($calls as $call) {
             foreach ($call->args as $position => $arg) {
                 if ($this->shouldSkipArg($arg)) {
-                    return [];
+                    continue;
```

Comment:
`return []` was right, when there is a first class callable or unpack in list of  calls, should be immediatelly stop all together since we don't know what the arg passed allowed.

I will create new PR for it.

Reply:
I created PR 

- https://github.com/rectorphp/rector-src/pull/7358

for it.